py/pysparkling/initializer.py

```diff
@@ -118,11 +108,18 @@ def __get_sw_jar():
             return os.path.abspath(resource_filename("sparkling_water", 'sparkling_water_assembly.jar'))
 
     @staticmethod
-    def __find_spark_cl(start_cl, cl_name):
-        cl = start_cl
+    def __get_logger(jvm):
+        return jvm.org.apache.log4j.LogManager.getLogger("org")
+
+    @staticmethod
+    def __add_url_to_classloader(jvm, url):
+        cl = jvm.Thread.currentThread().getContextClassLoader()
         while cl:
-            name = cl.getClass().getName()
-            if (name == cl_name):
-                return cl
+            methods = [m.getName() for m in cl.getClass().getDeclaredMethods()]
+            if "addURL" in methods:
+                cl.addURL(url)
+                Initializer.__get_logger(jvm).info("Adding " + url.toString() + " to classloader"
```

Comment:
nit: Would it be possible to use some string interpolation (E.g. `.format(...)`) and have the message at one line?

Reply:
Good idea, done!